src/libServer/IsolatedServer.cpp

```diff
@@ -443,6 +443,52 @@ bool IsolatedServer::ValidateTxn(const Transaction& tx, const Address& fromAddr,
   return true;
 }
 
+bool IsolatedServer::ValidateEthTxn(const Transaction& tx,
+                                    const Address& fromAddr,
+                                    const Account* sender) const {
+  if (DataConversion::UnpackA(tx.GetVersion()) != CHAIN_ID) {
+    throw JsonRpcException(
+        ServerBase::RPC_VERIFY_REJECTED,
+        std::string("CHAIN_ID incorrect: ") +
+            std::to_string(DataConversion::UnpackA(tx.GetVersion())) +
+            " when expected " + std::to_string(CHAIN_ID));
+  }
+
+  if (tx.GetCode().size() > MAX_EVM_CONTRACT_SIZE_BYTES) {
+    throw JsonRpcException(ServerBase::RPC_VERIFY_REJECTED,
+                           "Code size is too large");
+  }
+
+  if (tx.GetGasPrice() < MIN_ETH_GAS_PRICE_IN_WEI) {
+    throw JsonRpcException(ServerBase::RPC_VERIFY_REJECTED,
+                           "Gas price is too low");
+  }
+
+  if (!Validator::VerifyTransaction(tx)) {
+    throw JsonRpcException(ServerBase::RPC_VERIFY_REJECTED,
+                           "Unable to verify transaction");
+  }
+
+  if (IsNullAddress(fromAddr)) {
+    throw JsonRpcException(ServerBase::RPC_INVALID_ADDRESS_OR_KEY,
+                           "Invalid address for issuing transactions");
+  }
+
+  if (sender == nullptr) {
+    throw JsonRpcException(ServerBase::RPC_INVALID_ADDRESS_OR_KEY,
+                           "The sender of the txn has no balance");
```

Comment:
The message doesn't correspond to the code.

Reply:
Yep, will fix.